### PR TITLE
Checkout shortcode additional parameters

### PR DESCRIPTION
**Pull request by SebaSOFT**

### CODE REVIEW

File: plenigo_plugin/PlenigoShortcodeManager.php [hunk 1]
```diff
@@ -155,13 +155,19 @@ public function plenigo_handle_shortcode($atts, $content = null, $tag = null) {
...
+            'source' => "",
+            'target' => "",
+            'affiliate' => ""
                 ), $atts);
 
         $btnTitle = $a['title'];
         $cssClass = $a['class'];
         $prodId = $a['prod_id'];
         $regCheck = $a['register'];
+        $sourceURL = $a['soiurce'];
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoShortcodeManager.php]: Missspelling soiurce

File: plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php [hunk 2]
```diff
@@ -67,16 +70,47 @@ public function build($settings = array(), $loginToken = null, $showRegisterFirs
...
+            $strFirstParam = ", '" . $loginToken . "'";
+        }
+        $strSourceURL = "";
+        $strTargetURL = "";
+        if (!is_null($sourceUrl)) {
+            PlenigoManager::get()->notice($clazz, "Source URL:" . $sourceUrl);
+            $strSourceURL = ", '" . $sourceUrl . "'";
+            if (!$showRegisterFirst && is_null($loginToken)) {
+                $strFirstParam = ", false";
+            }
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php:91]: Is this missing here?
if (is_null($targetUrl)) {
                $targetUrl = ", null";
}

File: plenigo_sdk/plenigo/models/FailedPayment.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+     * Setter method for date.
+     * 
+     * @param string $date
+     */
+    public function setDate($date) {
+        $this->date = $date;
+    }
+
+    /**
+     * Setter method customer id.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPayment.php:110]: Setter method for customer id.

File: plenigo_sdk/plenigo/models/FailedPayment.php [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+     * Setter method for product id.
+     * 
+     * @param string $productId
+     */
+    public function setProductId($productId) {
+        $this->productId = $productId;
+    }
+
+    /**
+     * Setter method product title.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPayment.php]: Setter method for product title.

File: plenigo_sdk/plenigo/models/FailedPayment.php [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+     * 
+     * @param string $productTitle
+     */
+    public function setProductTitle($productTitle) {
+        $this->productTitle = $productTitle;
+    }
+
+    /**
+     * Setter method status.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPayment.php:137]: Setter method for status.

File: plenigo_sdk/plenigo/models/FailedPayment.php [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        $instance = new FailedPayment();
+        $instance->setDate(isset($map['date']) ? $map['date'] : null);
+        $instance->setCustomerId(isset($map['customerId']) ? $map['customerId'] : null);
+        $instance->setProductId(isset($map['productId']) ? $map['productId'] : null);
+        $instance->setProductTitle(isset($map['title']) ? $map['title'] : null);
+        $instance->setStatus(isset($map['status']) ? $map['status'] : null);
+
+        return $instance;
+    }
+
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPayment.php]: Empty line

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        if (is_array($array)) {
+            $this->elements = $array;
+        }
+        $this->pageNumber = $pageNumber;
+        $this->size = $size;
+        $this->totalElements = $totalElements;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: Get the page number

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+     * 
+     * @return int
+     */
+    public function getPageNumber() {
+        return $this->pageNumber;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: Get the page size

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+     * 
+     * @return int
+     */
+    public function getSize() {
+        return $this->size;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: Get all elements

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+     * 
+     * @return array
+     */
+    public function getElements() {
+        return $this->elements;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: Get total elements

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+     * @return FailedPaymentList instance.
+     */
+    public static function createFromMap(array $map) {
+        $pageNumber = isset($map['pageNumber']) ? $map['pageNumber'] : 0;
+        $size = isset($map['size']) ? $map['size'] : 10;
+        $totalElements = isset($map['totalElements']) ? $map['totalElements'] : 0;
+
+        $arrElements = isset($map['elements']) ? $map['elements'] : array();
+        $arrResulting = array();
+        foreach ($arrElements as $cpnyUser) {
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: $cpnyUser?

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+            $user = FailedPayment::createFromMap((array) $cpnyUser);
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: $user?

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Creates a FailedPaymentList instance from an array of maps.
+     * 
+     * @param array $userArray The array of maps to use for the instance creation.
+     * 
+     * @return \plenigo\models\FailedPaymentList instance.
+     */
+    public static function createFromArray(array $userArray) {
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: userArray?

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+        $pageNumber = 0;
+        $size = count($userArray);
+        $totalElements = count($userArray);
+        $arrResulting = array();
+        foreach ($userArray as $cpnyUser) {
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: $cpnyUser?

File: plenigo_sdk/plenigo/models/FailedPaymentList.php [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+            $user = FailedPayment::createFromMap((array) $cpnyUser);
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/FailedPaymentList.php]: $user

File: plenigo_sdk/plenigo/models/Order.php [hunk 16]
```diff
@@ -0,0 +1,346 @@
...
+
+require_once __DIR__ . '/OrderItem.php';
+
+use \plenigo\models\OrderItem;
+
+/**
+ * Order
+ * 
+ * <p>
+ * This class contains the failed payment information.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Wrong comment

File: plenigo_sdk/plenigo/models/Order.php [hunk 17]
```diff
@@ -0,0 +1,346 @@
...
+    private $orderItems;
+
+    /**
+     * Default constructor.
+     */
+    private function __construct() {
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the order id

File: plenigo_sdk/plenigo/models/Order.php [hunk 18]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return string
+     */
+    public function getOrderId() {
+        return $this->orderId;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the order date

File: plenigo_sdk/plenigo/models/Order.php [hunk 19]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return string
+     */
+    public function getOrderDate() {
+        return $this->orderDate;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the customer id

File: plenigo_sdk/plenigo/models/Order.php [hunk 20]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return string
+     */
+    public function getCustomerId() {
+        return $this->customerId;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the currency

File: plenigo_sdk/plenigo/models/Order.php [hunk 21]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return string
+     */
+    public function getCurrency() {
+        return $this->currency;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the cumulated price

File: plenigo_sdk/plenigo/models/Order.php [hunk 22]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return number
+     */
+    public function getCumulatedPrice() {
+        return $this->cumulatedPrice;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the vat number

File: plenigo_sdk/plenigo/models/Order.php [hunk 23]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return string
+     */
+    public function getVatNumber() {
+        return $this->vatNumber;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the shipping costs

File: plenigo_sdk/plenigo/models/Order.php [hunk 24]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return number
+     */
+    public function getShippingCosts() {
+        return $this->shippingCosts;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the shipping costs taxes

File: plenigo_sdk/plenigo/models/Order.php [hunk 25]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return number
+     */
+    public function getShippingCostsTaxes() {
+        return $this->shippingCostsTaxes;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the purchase order indicator

File: plenigo_sdk/plenigo/models/Order.php [hunk 26]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return number
+     */
+    public function getPurchaseOrderIndicator() {
+        return $this->purchaseOrderIndicator;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the discount

File: plenigo_sdk/plenigo/models/Order.php [hunk 27]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return number
+     */
+    public function getDiscount() {
+        return $this->discount;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the discount percentage

File: plenigo_sdk/plenigo/models/Order.php [hunk 28]
```diff
@@ -0,0 +1,346 @@
...
+     * 
+     * @return int
+     */
+    public function getDiscountPercentage() {
+        return $this->discountPercentage;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php]: Get the order items

File: plenigo_sdk/plenigo/models/Order.php [hunk 29]
```diff
@@ -0,0 +1,346 @@
...
+     * Setter method for shipping costs taxes.
+     * 
+     * @param number $shippingCostsTaxes
+     */
+    public function setShippingCostsTaxes($shippingCostsTaxes) {
+        $this->shippingCostsTaxes = $shippingCostsTaxes;
+    }
+
+    /**
+     * Setter method purchased order indicator.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/Order.php:282]: Setter method for purchased order indicator.

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 30]
```diff
@@ -0,0 +1,305 @@
+<?php
+
+namespace plenigo\models;
+
+/**
+ * OrderItem
+ * 
+ * <p>
+ * This class contains the failed payment information.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php:9]: Wrong comment

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 31]
```diff
@@ -0,0 +1,305 @@
...
+    private $status;
+
+    /**
+     * Default constructor.
+     */
+    private function __construct() {
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the product id

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 32]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return string
+     */
+    public function getProductId() {
+        return $this->productId;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the title

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 33]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return string
+     */
+    public function getTitle() {
+        return $this->title;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the price

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 34]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return number
+     */
+    public function getPrice() {
+        return $this->price;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the shipping costs

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 35]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return number
+     */
+    public function getShippingCosts() {
+        return $this->shippingCosts;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the taxes

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 36]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return number
+     */
+    public function getTaxes() {
+        return $this->taxes;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the tax country

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 37]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return string
+     */
+    public function getTaxCountry() {
+        return $this->taxCountry;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get quantity

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 38]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return int
+     */
+    public function getQuantity() {
+        return $this->quantity;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the cost center

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 39]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return string
+     */
+    public function getCostCenter() {
+        return $this->costCenter;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the revenue account

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 40]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @return string
+     */
+    public function getRevenueAccount() {
+        return $this->revenueAccount;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Get the status

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 41]
```diff
@@ -0,0 +1,305 @@
...
+     * Setter method for price.
+     * 
+     * @param number $price
+     */
+    public function setPrice($price) {
+        $this->price = $price;
+    }
+
+    /**
+     * Setter method shipping costs.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Setter method for shipping costs.

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 42]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @param number $shippingCosts
+     */
+    public function setShippingCosts($shippingCosts) {
+        $this->shippingCosts = $shippingCosts;
+    }
+
+    /**
+     * Setter method taxes.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Setter method for taxes.

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 43]
```diff
@@ -0,0 +1,305 @@
...
+     * 
+     * @param number $taxes
+     */
+    public function setTaxes($taxes) {
+        $this->taxes = $taxes;
+    }
+
+    /**
+     * Setter method tax country.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Setter method for taxes.

File: plenigo_sdk/plenigo/models/OrderItem.php [hunk 44]
```diff
@@ -0,0 +1,305 @@
...
+        $instance->setShippingCosts(isset($map['shippingCosts']) ? $map['shippingCosts'] : null);
+        $instance->setTaxes(isset($map['taxes']) ? $map['taxes'] : null);
+        $instance->setTaxCountry(isset($map['taxCountry']) ? $map['taxCountry'] : null);
+        $instance->setQuantity(isset($map['quantity']) ? $map['quantity'] : null);
+        $instance->setCostCenter(isset($map['costCenter']) ? $map['costCenter'] : null);
+        $instance->setRevenueAccount(isset($map['revenueAccount']) ? $map['revenueAccount'] : null);
+        $instance->setStatus(isset($map['status']) ? $map['status'] : null);
+        return $instance;
+    }
+
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderItem.php]: Empty line

File: plenigo_sdk/plenigo/models/OrderList.php [hunk 45]
```diff
@@ -0,0 +1,126 @@
...
+        if (is_array($array)) {
+            $this->elements = $array;
+        }
+        $this->pageNumber = $pageNumber;
+        $this->size = $size;
+        $this->totalElements = $totalElements;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderList.php:48]: Get the page number

File: plenigo_sdk/plenigo/models/OrderList.php [hunk 46]
```diff
@@ -0,0 +1,126 @@
...
+     * 
+     * @return int
+     */
+    public function getPageNumber() {
+        return $this->pageNumber;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderList.php:57]: Get the page size

File: plenigo_sdk/plenigo/models/OrderList.php [hunk 47]
```diff
@@ -0,0 +1,126 @@
...
+     * 
+     * @return int
+     */
+    public function getSize() {
+        return $this->size;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderList.php:66]: Get the elements

File: plenigo_sdk/plenigo/models/OrderList.php [hunk 48]
```diff
@@ -0,0 +1,126 @@
...
+     * 
+     * @return array
+     */
+    public function getElements() {
+        return $this->elements;
+    }
+
+    /**
+     * Getter method.
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderList.php:75]: Get the total elements

File: plenigo_sdk/plenigo/models/OrderList.php [hunk 49]
```diff
@@ -0,0 +1,126 @@
...
+
+        $arrResulting = array();
+        foreach ($orderArray as $cpnyOrder) {
+            $order = Order::createFromMap((array) $cpnyOrder);
+            array_push($arrResulting, $order);
+        }
+
+        return new OrderList($arrResulting, $pageNumber, $size, $totalElements);
+    }
+
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/models/OrderList.php]: Empty line

File: plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php (hunk 2 above)
Comment (SebaSOFT) [reply plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php:91]: nope, source and target url are handled later

File: plenigo_sdk/plenigo/models/FailedPayment.php (hunk 3 above)
Comment (SebaSOFT) [inline plenigo_sdk/plenigo/models/FailedPayment.php]: That's what it says. Please explain

File: plenigo_sdk/plenigo/models/FailedPayment.php (hunk 4 above)
Comment (SebaSOFT) [inline plenigo_sdk/plenigo/models/FailedPayment.php]: That's what it says. Please explain

File: plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php (hunk 2 above)
Comment (robertholfeld) [reply plenigo_sdk/plenigo/builders/CheckoutSnippetBuilder.php:91]: But when $targetUrl and $affiliatedId is null where is it handled then?

File: plenigo_sdk/plenigo/models/FailedPaymentList.php (hunk 11 above)
Comment (SebaSOFT) [reply plenigo_sdk/plenigo/models/FailedPaymentList.php]: renamed

File: plenigo_sdk/plenigo/models/FailedPaymentList.php (hunk 12 above)
Comment (SebaSOFT) [reply plenigo_sdk/plenigo/models/FailedPaymentList.php]: renamed

**Pull request merged**